### ISSUE TITLE
Data Schema for Existence and Expiration...

### ISSUE THREAD

**Issue by azadag** [antievictionmappingproject/covid-19-map#107]
Data Schema for Existence and Expiration

This would be for use on the map and the the pop ups, Brett is working on some designs and Chris has some language implemented when a specific expiration date exists (but we also will want to indicate if a specific protection exists, NOT the same thing)

![image](https://user-images.githubusercontent.com/987876/84813323-a7176300-afc4-11ea-981a-44002ae1b48f.png)

﻿
1. There are three types (language can change but we are going with) "Legislative/Executive Eviction Policies, "Court-Based Eviction Policies", and "Rent Relief Policies"...

2. And any combination of those three can exist for a location along with expirations for those policies. These can be specific dates, dates tied to states of exemptions (eg 90 days which we are attempting to track), or open-ended policies. 

So in the popups, we want to express information about the existence of these different types of policies. We have made a decision to use cross-hatching if ANY type of policy at the jurisdiction level. And the popup now populates based on 4 dates passed through

we currently have 4 strict date/time fields
 | end_date_earliest  | end_date_legist  | end_date_rent_relief  | end_date_court
along with a TRUE/FALSE  | has_expired_protections

We need to supplement the end_dates_ with variable or variables to indicate both the existence of a type of policy and to also relay information about open-ended or linked to states of exemption.

**TLDR SCHEMA ASK:**
1. how to show existence or non existence of of 3 types of protections. 
each has 4 possible states: 

a. never implemented
b. current
c. expired
d. expiring soon

2. How to track additional dates of expiry + open-ended + States of Exemption

Related States of Exemptions - Complication
We can either just relay the existence of each state of exemption law, until we know it has passed and then label it as expired. Or to be more pro-active we have thought about having the research team create a data-base of states of exemptions and with that create changing but hard dates (this would be more labor intensive for research / rely on creating processes for people to update continually during office hours)

**Comment by clhenrick** [comment]
Thanks @azadag for laying this out very concisely. 

For the first schema ask, we could show an un-ordered list of the protection types in the infowindow under a sub heading, for example it could look like:

### Types of Protection Policies:
- **Legislative/Executive Eviction:** Currently in effect.
- **Court-Based Eviction:** Not applicable.
- **Rent Relief:** Expiring soon.

or perhaps just make each one a label similar to how the other data fields are presented:

**Legislative/Executive Eviction Policies:** Currently in effect.  
**Court-Based Eviction Policies:** Not applicable.  
**Rent Relief Policies:** Expiring soon.

I used "not applicable" above as it seems more clear to me then "never implemented" but I'm for whatever everyone else agrees on. In regards to the language / wording choice, I would think about who the target audience is, and probably how to keep it straightforward and as few words as possible.

The data could be structured as 3 different columns, e.g. 
- `policy_legist_status`
- `policy_court_status`
- `policy_rent_relief_status`

and each column would be an enumerable containing one of the four statuses you mentioned: "never implemented", "current," "expired", or "expiring soon". 

We would want to stabilize the language for any of these names that are displayed in the infowindow so that we don't ask our translators helping with i18n to repeat their work.

For the 2nd ask, it sounds a bit more complicated. I would vote for the simpler option, relaying the existence of the policy and then label it as expired once it has expired. I would hate to put that much more work on the research team (maintaining another dataset), but it's of course up to you all to decide on.

Happy to discuss any of this more 1x1 if that would help!
